x-pack/plugins/lens/public/indexpattern_plugin/dimension_panel.test.tsx

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mount, shallow } from 'enzyme';
+import React from 'react';
+import { EuiComboBox } from '@elastic/eui';
+import { IndexPatternPrivateState } from './indexpattern';
+import { getColumnOrder, getPotentialColumns } from './operations';
+import { IndexPatternDimensionPanel } from './dimension_panel';
+
+jest.mock('./operations');
+
+const expectedIndexPatterns = {
+  1: {
+    id: '1',
+    title: 'my-fake-index-pattern',
+    timeFieldName: 'timestamp',
+    fields: [
+      {
+        name: 'timestamp',
+        type: 'date',
+        aggregatable: true,
+        searchable: true,
+      },
+      {
+        name: 'bytes',
+        type: 'number',
+        aggregatable: true,
+        searchable: true,
+      },
+      {
+        name: 'source',
+        type: 'string',
+        aggregatable: true,
+        searchable: true,
+      },
+    ],
+  },
+};
+
+describe('IndexPatternDimensionPanel', () => {
+  let state: IndexPatternPrivateState;
+
+  beforeEach(() => {
+    state = {
+      indexPatterns: expectedIndexPatterns,
+      currentIndexPatternId: '1',
+      columnOrder: ['col1'],
+      columns: {
+        col1: {
+          operationId: 'op1',
+          label: 'Value of timestamp',
+          dataType: 'date',
+          isBucketed: false,
+
+          // Private
+          operationType: 'value',
+          sourceField: 'timestamp',
+        },
+      },
+    };
+
+    jest.clearAllMocks();
+  });
+
+  it('should render a dimension panel', () => {
+    const wrapper = shallow(
+      <IndexPatternDimensionPanel
+        state={state}
+        setState={() => {}}
+        columnId={'col2'}
+        filterOperations={() => true}
+      />
+    );
+
+    expect(wrapper).toMatchSnapshot();
```

Comment:
I'd like to get some feedback from @chrisdavies and @flash1293 about whether this snapshot test is adding value to your workflow. It got in my way a lot, and the snapshot is quite large, so it feels like it might be worth eliminating.

Reply:
I agree that this snapshot isn't really adding anything meaningful the tests below don't capture already.

Reply:
Removing